### DISCUSSION TITLE
Mantine NavLink with Remix NavLink

### DISCUSSION THREAD

**Discussion by timmclean** [mantinedev/mantine#5461] [Q&A] [answered]
Normally when I use the NavLink component from Remix, it automatically becomes "active" when the current page URL matches the value of the `to` prop. Is there a way to get this behaviour with Mantine's NavLink component?

I read about `renderRoot`. I could use that to render Mantine's NavLink as a Remix NavLink. But I don't know how to get the "active" state up from the Remix NavLink to the Mantine NavLink so that I can take advantage of Mantine's "active" styling.

**Comment by bhuynhdev** [comment, marked as answer]
Since May 11 2024, after PR https://github.com/mantinedev/mantine/pull/6180 is merged so that Mantine can apply `active` styles to NavLinks with `aria-current='page'`. this has been quite easy

This is because [Remix Navlink auto add `aria-current='page'` to active nav links](https://remix.run/docs/en/main/components/nav-link#aria-current), and thus will automatically trigger Mantine's styling. So you just need to `renderRoot` to Remix NavLink component, and specify the `to` prop

```tsx
// Simply use RemixNavLink for `renderRoot`.
// Don't even need to include MantineNavLink's `href` prop. Just need the `to` in RemixNavLink
<MantineNavLink label="Home page" renderRoot={(props) => <RemixNavLink to='/page' {...props} />} />

// You can even make it into a component
/**
 * Extend Mantine's NavLink to render as a RemixNavLink, so the "active" state from Remix is connected to Mantine's style
 **/
import { NavLinkProps as MantineNavLinkProps, ElementProps, ...} from "@mantine/core"
 
function MyNavLink(props: MantineNavLinkProps & ElementProps<"a", keyof MantineNavLinkProps>) {
  return (
    <MantineNavLink {...props} renderRoot={({className}) => <RemixNavLink to={props.href} className={className} />}>
      {props.children}
    </MantineNavLink>
  )
}

<MyNavLink href="/page" label="Home page" />
```

**Reply by corydeppen** [reply]
@bhuynhdev I don't believe the `href` prop is needed on the Mantine `NavLink` when a Remix `NavLink` is being rendered as the root element. Here's a [quick example](https://stackblitz.com/edit/remix-run-remix-qdbpkl?file=app%2Fcomponents%2Fsite-nav.tsx).

**Reply by bhuynhdev** [reply]
Oh that's even better. Thanks. I have updated my comment to reflect this tip.
